### PR TITLE
London 9 Olus Odude JS CORE2 WK1 

### PR DESCRIPTION
**Pull request by OlushoreOdude**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name: Olus
- Your City: London
- Your Slack Name: olushoreodude

# Homework Details

- Module: jscore2
- Week: wk1

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

**Comment by OlushoreOdude** [comment]
Hi @maxf, i will run some local tests to fix checks.

File: 1-exercises/A-accessing-values/exercise2.js [hunk 1]
```diff
@@ -7,7 +7,7 @@
 let capitalCities = {
   UnitedKingdom: "London",
   China: "Beijing",
-  Peru: "Lima"
+  Peru: "Lima",
```
Comment (maxf) [inline 1-exercises/A-accessing-values/exercise2.js:10]: It also works without the comma. Both are the same.

File: 1-exercises/A-accessing-values/exercise3.js [hunk 2]
```diff
@@ -21,12 +21,36 @@ let basketballTeam = {
 */
 
 // write code here
+function getAcessAndSort(object, key2sort) {
+  //run a for in loop for array keys
```
Comment (maxf) [inline 1-exercises/A-accessing-values/exercise3.js:25]: where's the "for in loop"?

File: 1-exercises/A-accessing-values/exercise3.js [hunk 3]
```diff
@@ -21,12 +21,36 @@ let basketballTeam = {
...
+  //if array key is eqal to basket ball players
+  let keyValue = object[`${key2sort}`];
```
Comment (maxf) [inline 1-exercises/A-accessing-values/exercise3.js:27]: A bit convoluted :) 
How about just `object[key2sort]` ?

File: 1-exercises/A-accessing-values/exercise3.js [hunk 4]
```diff
@@ -21,12 +21,36 @@ let basketballTeam = {
 */
 
 // write code here
+function getAcessAndSort(object, key2sort) {
```
Comment (maxf) [inline 1-exercises/A-accessing-values/exercise3.js:24]: It's a good idea to generalise the code in order to make it work for future scenarios. But in this case you don't need to, so it's ok not to have `key2sort` and the `Array.isArray` check. You're allowed to assume the input object is valid.
However in a professional context this is a good thing to do, in moderation.

File: 1-exercises/A-accessing-values/exercise3.js [hunk 5]
```diff
@@ -21,12 +21,36 @@ let basketballTeam = {
...
+  //run a for in loop for array keys
+  //if array key is eqal to basket ball players
+  let keyValue = object[`${key2sort}`];
+  console.log(keyValue);
```
Comment (maxf) [inline 1-exercises/A-accessing-values/exercise3.js:28]: This looks like debugging code you've not removed at the end

File: 1-exercises/A-accessing-values/exercise3.js [hunk 6]
```diff
@@ -21,12 +21,36 @@ let basketballTeam = {
...
+    return `the key ${key2sort} is not an array`;
+    //keyValue.sort()
+  }
+  keyValue.sort();
+  //console.log(KeyValue);
+  // please explain why console.log breaks code.
+  //is for loop each item console logs more efficient than map
+  //as it does not return an array?
 
+  return keyValue.map((el) => {
```
Comment (maxf) [inline 1-exercises/A-accessing-values/exercise3.js:43]: Either you use `map` to create an array of values that you can print in one call at the end, or you use a `forEach` within which you can print each value.
Here you don't need to use `map` as you're not building an array

File: 1-exercises/B-setting-values/exercise1.js [hunk 7]
```diff
@@ -22,8 +22,11 @@ let capitalCities = {
     - Add a population of 9750000 to Peru's capital city.
 */
 
-// write code here
-
+capitalCities.UnitedKingdom.population = 8980000;
+capitalCities.China.population = 21500000;
+capitalCities.Peru = {};
+capitalCities.Peru.name = "lima";
+capitalCities.Peru.population = 9750000;
```
Comment (maxf) [inline 1-exercises/B-setting-values/exercise1.js:29]: Excellent. You can also write
`capitalCities.Peru = { name: "lima", population: 9750000 }`

File: 1-exercises/C-undefined-properties/exercise.js [hunk 8]
```diff
@@ -36,3 +38,5 @@ let myPet = {
 };
 
 console.log(myPet.getName());
+// does not automatically return string as it is not on the same line as the function keyword
```
Comment (maxf) [inline 1-exercises/C-undefined-properties/exercise.js:41]: That is not why it doesn't return the string.

File: 1-exercises/C-undefined-properties/exercise.js [hunk 9]
```diff
@@ -36,3 +38,5 @@ let myPet = {
...
+//as a result returns undefined as that is the standard return value of a function
```
Comment (maxf) [inline 1-exercises/C-undefined-properties/exercise.js:42]: or in simpler terms, the function is missing `return` before the string.

File: 1-exercises/D-object-methods/exercise.js [hunk 10]
```diff
@@ -9,6 +9,9 @@
 
 let student = {
   // write code here
+   getName: function(name) {
+    return `Student name: ${name}`;
+  }
```
Comment (maxf) [inline 1-exercises/D-object-methods/exercise.js:14]: excellent!

File: 2-mandatory/1-recipes.js [hunk 11]
```diff
@@ -22,4 +22,17 @@
   You should write and log at least 5 recipes
 */
 
-// write code here
\ No newline at end of file
+// write code here
+
+let recipe = {
```
Comment (maxf) [inline 2-mandatory/1-recipes.js:27]: only 1? We wanted 5 :)

File: 2-mandatory/1-recipes.js [hunk 12]
```diff
@@ -22,4 +22,17 @@
...
+
+  title: "cake",
+  servings: 5,
+  ingredients:["milk", "eggs", "butter", "flour", "sugar"]
+  }
+
+const { title, servings, ingredients } = recipe;
```
Comment (maxf) [inline 2-mandatory/1-recipes.js:34]: Using variables here is a bit overkill, as you can just write below
`console.log(recipe.title)`

File: 2-mandatory/2-currency-code-lookup.js [hunk 13]
```diff
@@ -18,7 +18,18 @@ const COUNTRY_CURRENCY_CODES = [
 ];
 
 function createLookup(countryCurrencyCodes) {
-  // write code here
+  let obj = {};
+countryCurrencyCodes.forEach((elem) => {
```
Comment (maxf) [inline 2-mandatory/2-currency-code-lookup.js:22]: please fix the indentation

File: 2-mandatory/2-currency-code-lookup.js [hunk 14]
```diff
@@ -18,7 +18,18 @@ const COUNTRY_CURRENCY_CODES = [
...
+    obj[`${elem[0]}`] = elem[1]
```
Comment (maxf) [inline 2-mandatory/2-currency-code-lookup.js:23]: or `obj[elemen[0]] = elem[1]`

File: 2-mandatory/2-currency-code-lookup.js [hunk 15]
```diff
@@ -18,7 +18,18 @@ const COUNTRY_CURRENCY_CODES = [
...
+});
+
+return obj;
+}
+
+let result = createLookup(COUNTRY_CURRENCY_CODES);
+
+
+console.log(result);
 }
```
Comment (maxf) [inline 2-mandatory/2-currency-code-lookup.js:33]: Something tells me you haven't run this code, or the test. It's got the wrong number of `}` or `{`

File: 2-mandatory/2-currency-code-lookup.js (hunk 13 above)
Comment (maxf) [inline 2-mandatory/2-currency-code-lookup.js:22]: please fix your indentation

File: 2-mandatory/1-recipes.js (hunk 11 above)
Comment (maxf) [inline 2-mandatory/1-recipes.js:27]: you should use `const` by default

File: 2-mandatory/3-shopping-list.js [hunk 16]
```diff
@@ -18,8 +18,21 @@ let pantry = {
...
+  // join all the items into one array using concat 
+  //filer the items that are not in the pantry
+  let obj = {};
+ let arrTest = recipe.ingredients.filter(el => {
+    return !pTestArr2.includes(el);
+    });
+  
+  obj.name = recipe.name;
+  obj.ingredients = arrTest;
+return obj;
```
Comment (maxf) [inline 2-mandatory/3-shopping-list.js:35]: indentation

File: 2-mandatory/4-restaurant.js [hunk 17]
```diff
@@ -19,9 +19,28 @@ const MENU = {
   falafel: 7.25,
 };
 
+
 let cashRegister = {
   // write code here
-}
+  orderBurger: function(balance){
```
Comment (maxf) [inline 2-mandatory/4-restaurant.js:25]: good

File: 2-mandatory/4-restaurant.js [hunk 18]
```diff
@@ -19,9 +19,28 @@ const MENU = {
...
+  if(balance >= MENU.burger){
```
Comment (maxf) [inline 2-mandatory/4-restaurant.js:26]: use a consistent style for white space. Here:
`if (balance >= MENU.burger) {`

File: 3-extra/1-count-words.js [hunk 19]
```diff
@@ -23,12 +23,33 @@
...
-
-  return wordCount;
+function countWords(string){
+ 
+  let test2S = string.split(" ");
+  let test3s = [...test2S];
+    
+    let result1 = test3s.map(el => {
+  
+  let count = 0;
```
Comment (maxf) [inline 3-extra/1-count-words.js:33]: indentation

File: 3-extra/1-count-words.js [hunk 20]
```diff
@@ -23,12 +23,33 @@
...
+let str = test2S;
+let query = el;
```
Comment (maxf) [inline 3-extra/1-count-words.js:35]: why is this variable needed?

File: 3-extra/1-count-words.js [hunk 21]
```diff
@@ -23,12 +23,33 @@
...
 
-function countWords(string) {
-  const wordCount = {};
-
-  // write code here
-
-  return wordCount;
+function countWords(string){
+ 
+  let test2S = string.split(" ");
```
Comment (maxf) [inline 3-extra/1-count-words.js:28]: maybe try and find simpler names for your variables when your can, even if they are longer. Here you could call it `words`.
